tests/test_figure_resampler.py

```diff
@@ -296,7 +296,7 @@ def test_log_axis():
         assert len(out) == 2
         assert (x1 - x0) < 10
         assert len(out[1]["x"]) == 1000
-        assert out[-1]["x"][0] >= 100
+        assert out[-1]["x"][0] >= 99
```

Comment:
(maybe because of a small rounding error with log?)

Reply:
You are right!
-> end_idx = bisect.bisect_right(hf_trace_data["x"], end)
-> end == 49999.99999999999

Nevertheless alter boundary. It would benefit readability